From 68bf659c73675ecd3f462fd7ae6951787d389900 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:34 -0400
Subject: [ethernet] i40e: Redistribute queue vectors after DCB reconfiguration

Message-id: <1409658042-22689-94-git-send-email-sassmann@redhat.com>
Patchwork-id: 90288
O-Subject: [RHEL7.1 PATCH 093/281] i40e: Redistribute queue vectors after DCB reconfiguration
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Neerav Parikh <Neerav.Parikh@intel.com>

The queue vectors are originally distributed among the total
number of queues that are configured for a given VSI. The number
of queues that are configured for a VSI depends on the number of
TCs that are enabled for the VSI at VSI setup/creation time.

But, the total number of queues that are configured for a VSI may
change based on change in the total number of TCs enabled as a
result of change in DCB configuration based on DCBX.

Hence, it is required for the queue vectors to be redistributed
when the total queues that are configured for a given VSI changes.

Without redistributing the vectors the queues that are configured
for the newly enabled TCs may not be able to do any Tx/Rx.

Change-ID: I4e780903db019e6199c7ce43627cef22c916acfe
Signed-off-by: Neerav Parikh <Neerav.Parikh@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 0672a0918374036b7dffd9e60eeb63213e994f1b)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 5d8bde5..46dc21a 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -4024,6 +4024,8 @@ static void i40e_dcb_reconfigure(struct i40e_pf *pf)
      pf->vsi[v]->seid);
    /* Will try to configure as many components */
   } else {
+   /* Re-configure VSI vectors based on updated TC map */
+   i40e_vsi_map_rings_to_vectors(pf->vsi[v]);
    if (pf->vsi[v]->netdev)
     i40e_dcbnl_set_all(pf->vsi[v]);
   }
-- 
1.7.1